### ISSUE TITLE
Enhancement: [no-throw-literal] Specify allowed types

### ISSUE THREAD

**Issue by haines** [typescript-eslint/typescript-eslint#6226]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/no-throw-literal/

### Description

I would like to add an additional option to this rule, `allowThrowingTypes`, that would allow users to override the current hard-coded behaviour of checking for the global `Error` class.

I'd find this useful in Remix projects, where you can [throw a `Response` object](https://remix.run/docs/en/v1/route/loader#throwing-responses-in-loaders). Currently, I have to disable this rule, when what I would prefer to do is only allow `Error` or `Response` objects to be thrown.

I think this could potentially also be useful in codebases with custom error types, where you want to enforce usage of a more specific subtype of `Error`.

I'm not sure what the syntax of the option should be - you would need be able to specify the import location and type, rather than just matching on the type name. Are there any other rules where it is possible to precisely specify types in this way?

### Fail

```typescript
throw "/login";
```

### Pass

```typescript
// with allowThrowingTypes: ["Error", "Response"]
// redirect returns Response & { json(): Promise<never> }
throw redirect("/login");
```

**Comment by JoshuaKGoldberg** [comment]
Seems reasonable to me. We have an ongoing discussion about making a unified format for specifying these types: https://github.com/typescript-eslint/typescript-eslint/discussions/6017 -> https://github.com/eslint/eslint/discussions/16540. 

I'm hopeful those will resolve soon. But we should wait to take action on this feature request until then.

**Comment by raulmt** [comment]
Hi @JoshuaKGoldberg ! It seems a solution for specifying an allow list of types was implemented for another rule related to the mentioned discussion: https://github.com/typescript-eslint/typescript-eslint/pull/4436

Can that implementation unblock this issue now? (btw, I reached to this issue because of the same specific reason as @haines being able to throw Response :) )

**Comment by JoshuaKGoldberg** [comment]
Oh, yes! Thanks for the ping 🙂 I'd missed unblocking this issue. Marking as accepting PRs to use that format.

**Comment by kirkwaiblinger** [comment]
This is duplicated by https://github.com/typescript-eslint/typescript-eslint/issues/9525. Closing in favor of that issue.
